js/packages/deploy/deploy.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Nugget Industries
+ * 2017
+ *
+ * COMMAND LINE ARGUMENTS:
+ * -d | --debug:
+ *   runs bot in debug mode
+ * -l | --local:
+ *   runs bot the bot and the surface station in local mode (should be run with debug mode)
+ * --pi-address:
+ *   specify an address
+ * --start-surface:
+ *   starts the surface station too
+ */
+
+/*
+ * YE OLDE MONOLITHIC DEPLOYMENT SCRIPT
+ *
+ * The idea here is to have one big script that installs dependancies for the robot,
+ * copies everything in the 'remote' folder to the robot in an executeable place, and then
+ * executes it.
+ *
+ * Time will tell if this should be split up into separate scripts or not.
+ */
+
+// native dependanciea
+const { fork } = require('child_process');
+
+// package dependencies
+const yargs = require('yargs');
+const scp = require('scp2').scp;
+const remoteExec = require('remote-exec');
+
+// look for the pi here!
+const defaultPiAddress = 'spacenugget.local';
+
+const args = yargs
+    .usage('Usage: $0 [options]')
+    .version(false)
+    .option('d', {
+        alias: 'debug',
+        desc: 'use fake sensor values instead of real onez',
+        type: 'boolean'
+    })
+    .option('l', {
+        alias: 'local',
+        desc: 'run the robot on localhost and tell the surface to look there too',
+        type: 'boolean',
+    })
+    .option('P', {
+        alias: 'pi-address',
+        desc: 'connect to the robot at this address',
+        type: 'string',
+        default: defaultPiAddress,
+        nargs: 1
+    })
+    .option('S', {
+        alias: 'start-surface',
+        desc: 'start the surface station too! (ya lazy bum)',
+        type: 'boolean'
+    })
+    .option('N', {
+        alias: 'no-run',
+        desc: 'just copy the files, don\'t restart the nugget daemon on the pi',
+        type: 'boolean'
+    })
+    .alias('h', 'help')
+    .argv;
+
+const piPath = '/opt/rov2017';
+
+async function setupRobot(args) {
+    console.log('Starting robot setup');
+
+    /*
+     * if we're in debug mode, spawn botServer in debug mode
+     * resolve when the server sends an IPC message
+     */
+    ////////////////
+    // DEBUG MODE //
+    ////////////////
+    if (args.local) return new Promise(resolve => {
+        const botArgs = ['--local', '--debug'];
+        console.log('Starting robot in local and debug mode');
+
+        const forkOptions = {
+            execArgv: ['--inspect'],
+            stdio: ['pipe', 'pipe', 'pipe', 'ipc']
+        };
+        const child = fork(__dirname + '/../../remote/index.js', botArgs, forkOptions).on('message', () => {
+            console.log('Finished setting up robot in local and debug mode');
+            resolve(child);
+        });
+    });
+
+    ////////////////////
+    // NOT DEBUG MODE //
+    ////////////////////
+    console.log('Copying files to robot');
+    // copy files to robot, resolve when complete
+    const scpOptions = {
+        host: args.piAddress,
+        username: 'root',
```

Comment:
Defaulting to root considered bad practice, do we not care because we control both ends? Security isn't an issue for us to deal with, but do we want the automated stuff running around as root? It may not matter, just thought I'd bring it up.

Reply:
This is just because we're `scp`ing straight to `/opt` which the `pi` user doesn't have write access to. We can change write permissions for `/opt/rovsquad` so the `scp` doesn't need to happen as root.